stock_move_line_serial_unique/tests/test_stock_move_line_serial_unique.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.exceptions import ValidationError
+from odoo.tests.common import TransactionCase
+
+
+class TestStockMoveLineSerialUnique(TransactionCase):
+    def setUp(self):
+        super(TestStockMoveLineSerialUnique, self).setUp()
+        self.stock_location = self.env.ref("stock.stock_location_stock")
+        self.warehouse = self.env.ref("stock.warehouse0")
+        self.shelf1_location = self.env["stock.location"].create(
+            {
+                "name": "Test location",
+                "usage": "internal",
+                "location_id": self.stock_location.id,
+            }
+        )
+        self.vendor_location = self.env["stock.location"].create(
+            {
+                "name": "Test vendor location",
+                "usage": "supplier",
+                "location_id": self.stock_location.id,
+            }
+        )
```

Comment:
```suggestion
        vendor_location = self.env.ref("stock.stock_location_suppliers")
```
No need to start with `self.` if it's only used in the setup.